src/gadgets/num.rs

```diff
@@ -46,6 +47,50 @@ impl<Scalar: PrimeField> AllocatedNum<Scalar> {
         })
     }
 
+    /// Allocate a `Variable(Input)` in a `ConstraintSystem`.
+    pub fn alloc_input<CS, F>(mut cs: CS, value: F) -> Result<Self, SynthesisError>
+    where
+        CS: ConstraintSystem<Scalar>,
+        F: FnOnce() -> Result<Scalar, SynthesisError>,
+    {
+        let mut new_value = None;
+        let var = cs.alloc_input(
```

Comment:
If you look at the code for `inputize` it also adds a constraint. 
So I am wondering if this is correct and the `alloc + inputize` is just inefficient, or if there is more to it.